core-foundation/src/array.rs

```diff
@@ -92,11 +92,6 @@ impl<T> CFArray<T> {
         }
     }
 
-    #[deprecated(note = "please use `as_untyped` instead")]
-    pub fn to_untyped(self) -> CFArray {
```

Comment:
I realize now that my previous move to deprecate `to_untyped` and add `as_untyped` was not exactly making things more correct than before, if one should follow the API guidelines to the letter.
This removed `to_untyped` is a consuming convert (`owned -> owned (non-Copy types)`), so it should have been named `into_untyped` with that signature. So the old name was slightly "wrong".
On the other hand, the new `as_untyped` only borrows `&self`, but gives out an owned object (`borrowed -> owned (non-Copy types)`), so it should be named `to_untyped`, and not `as_` since that is for conversions `borrowed -> borrowed`.

Do you think I should change the name of the method that will be present in the `0.5.0` release? Or is that too much nitpicking and too much changing the API back and forth? If it should be changed this would be the correct signature:
```rust
pub fn to_untyped(&self) -> CFArray { ... }
```
In line with all the new `downcast_into` methods, we could also add a consuming one that avoids bumping the retain count:
```rust
pub fn into_untyped(self) -> CFArray { ... }
```

Reply:
Those seem like worthwhile changes.

Reply:
Check. I'll add a commit for that. I'll push it after my other PR is merged, as this one will need rebasing/manual conflict resolution after that one anyway.

Reply:
Done.